contrib/babelfishpg_tsql/src/tsqlIface.cpp

```diff
@@ -2447,6 +2481,14 @@ class tsqlBuilder : public tsqlCommonMutator
 					throw PGErrorWrapperException(ERROR, ERRCODE_FEATURE_NOT_SUPPORTED, 
 						format_errmsg("function %s does not exist", proc_name.c_str()), getLineAndPos(ctx));
 				}
+
+				if (pg_strcasecmp(proc_name.c_str(), "trim") == 0 
+					&& !(ctx->func_proc_name_server_database_schema()->server)
+					&& !(ctx->func_proc_name_server_database_schema()->database)
+					&& !(ctx->func_proc_name_server_database_schema()->schema))
+				{
+					rewritten_query_fragment.emplace(std::make_pair(ctx->func_proc_name_server_database_schema()->start->getStartIndex(), std::make_pair(::getFullText(ctx->func_proc_name_server_database_schema()), "sys.trim")));
+				}
```

Comment:
This could be shifted to above `/* built-in functions */` section under condition like `id->keyword()->TRIM()`, same for above `TSqlParserBaseListener` handling

Reply:
Done